### PR TITLE
Allow passing `compile_pymc` kwargs to sampling functions

### PR DESCRIPTION
**Pull request by ricardoV94**
This is needed for sampling of `RandomVariables` within `Scan`s among other things

### CODE REVIEW

File: pymc/sampling.py [hunk 1]
```diff
@@ -1746,13 +1746,16 @@ def sample_posterior_predictive(
     if size is not None:
         vars_to_sample = [change_rv_size(v, size, expand=True) for v in vars_to_sample]
 
+    if aesara_function_kwargs is None:
```
Comment (twiecki) [inline pymc/sampling.py]: Shouldn't this be `compile_kwargs`?

File: pymc/sampling.py [hunk 2]
```diff
@@ -1599,6 +1597,8 @@ def sample_posterior_predictive(
     idata_kwargs : dict, optional
         Keyword arguments for :func:`pymc.to_inference_data` if ``predictions=False`` or to
         :func:`pymc.predictions_to_inference_data` otherwise.
+    compile_kwargs: dict, optional
```
Comment (twiecki) [inline pymc/sampling.py:1599]: Or this be `aesara_function_kwargs`

File: pymc/sampling.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/sampling.py]: yeah, typo when I changed the name

File: pymc/sampling.py [hunk 3]
```diff
@@ -1746,13 +1745,16 @@ def sample_posterior_predictive(
     if size is not None:
         vars_to_sample = [change_rv_size(v, size, expand=True) for v in vars_to_sample]
 
+    if compile_kwargs is None:
```
Comment (twiecki) [inline pymc/sampling.py:1748]: Seems like the tests didn't catch that.

Comment (ricardoV94) [reply pymc/sampling.py:1748]: What didn't they catch? There were a bunch of failing tests. Is there an error still?

Comment (twiecki) [reply pymc/sampling.py:1748]: I didn't check, but did the test fail before you fixed the wrong name or did you add a new one that would have caught it?

Comment (ricardoV94) [reply pymc/sampling.py:1748]: The tests (new and old) failed. I just didn't rerun all of them locally after doing the variable name change.

**Review by twiecki — APPROVED** [review]

**Pull request merged**